[nrfconnect] Changes in documentation (#1)
LipinskiPNordicSemi/chip-fork: examples/lock-app/nrfconnect/README.md

        $ git submodule update --init

> **Note**:
>
> For Linux operating system
> Install [SEGGER J-Link Software](https://www.segger.com/downloads/jlink/#J-LinkSoftwareAndDocumentationPack).

### Installing command line requirements

To build on the command line, complete the following steps:

-   With admin permissions enabled, download and install the [nRF Command Line Tools](https://www.nordicsemi.com/Products/Development-tools/nrf-command-line-tools).

### Install Toolchain Manager

Toolchain Manager is available from [nRF Connect for Desktop](https://www.nordicsemi.com/Products/Development-tools/nrf-connect-for-desktop), a cross-platform tool that provides different applications that simplify installing the nRF Connect SDK. Both the tool and the application are available for Windows, Linux, and macOS.

To install the Toolchain Manager app, complete the following steps:

1.  [Download nRF Connect for Desktop](https://www.nordicsemi.com/Products/Development-tools/nrf-connect-for-desktop/download#infotabs) for your operating system.

2.  Install and run the tool on your machine.

> 3.  In the **APPS** section, click `Install` next to Toolchain Manager.

Reviewer: ```suggestion
3.  In the **APPS** section, click **Install** next to Toolchain Manager.
```
I'm not sure if there are other guidelines for CHIP, so suggesting what we use for formatting in other Markdown docs.

### Install nRF Connect SDK

Complete the following steps to install the nRF Connect SDK source code:

1.  Open Toolchain Manager in nRF Connect for Desktop.

> 2.  Click the `Install` button next to the [recommended](../../../config/nrfconnect/.nrfconnect-recommended-revision) version of the nRF Connect SDK.

Reviewer: ```suggestion
2.  Click the **Install** button next to the [recommended](../../../config/nrfconnect/.nrfconnect-recommended-revision) version of the nRF Connect SDK.
```

3.  A pop-up window will inform you about the current installation directory. If you want to change the directory, click the

>     `Change directory` button. Otherwise, click the `Continue installation button`.

Reviewer: ```suggestion
    **Change directory** button. Otherwise, click the **Continue installation** button.
```

> 4.  When the nRF Connect SDK is installed on your machine, the Install button changes to the Open VS Code button.

Reviewer: ```suggestion
4.  When the nRF Connect SDK is installed on your machine, the **Install** button changes to the **Open VS Code** button.
```

> 5.  Click the dropdown menu next to the Open VS Code button for the installed nRF Connect SDK version, and select `Open terminal`.

Reviewer: ```suggestion
5.  Click the dropdown menu next to the Open VS Code button for the installed nRF Connect SDK version, and select **Open terminal**.
```

        $ git submodule update --init

> **Note**:
>
> For Linux operating system
> Install [SEGGER J-Link Software](https://www.segger.com/downloads/jlink/#J-LinkSoftwareAndDocumentationPack).

### Installing command line requirements

To build on the command line, complete the following steps:

> 1.  With admin permissions enabled, download and install the nRF [Command Line Tools](https://www.nordicsemi.com/Products/Development-tools/nrf-command-line-tools).

Reviewer 2: ```suggestion
1.  With admin permissions enabled, download and install the [nRF Command Line Tools](https://www.nordicsemi.com/Products/Development-tools/nrf-command-line-tools).
```

Reviewer 2: Btw. Do we need to use "1." here? It's just a single point.

Reviewer: Very good point. Don't really need to number a single step.

### Install Toolchain Manager

Toolchain Manager is available from [nRF Connect for Desktop](https://www.nordicsemi.com/Products/Development-tools/nrf-connect-for-desktop), a cross-platform tool that provides different applications that simplify installing the nRF Connect SDK. Both the tool and the application are available for Windows, Linux, and macOS.

To install the Toolchain Manager app, complete the following steps:

1.  [Download nRF Connect for Desktop](https://www.nordicsemi.com/Products/Development-tools/nrf-connect-for-desktop/download#infotabs) for your operating system.

2.  Install and run the tool on your machine.

3.  In the APPS section, click Install next to Toolchain Manager.

### Install nRF Connect SDK

Complete the following steps to install the nRF Connect SDK source code:

1.  Open Toolchain Manager in nRF Connect for Desktop.

> 2.  In SDK ENVIRONMENTS, click the Install button next to the [recomended](https://github.com/LipinskiPNordicSemi/chip-fork/blob/documentation/config/nrfconnect/.nrfconnect-recommended-revision) version of the nRF Connect SDK.

Reviewer 2: Can we use just a relative path to "config/nrfconnect/.nrfco..." - this is to avoid using external repository.

> 2.  In SDK ENVIRONMENTS, click the Install button next to the [recomended](https://github.com/LipinskiPNordicSemi/chip-fork/blob/documentation/config/nrfconnect/.nrfconnect-recommended-revision) version of the nRF Connect SDK.

Reviewer 2: also.. In SDK ENVIRONMENTS tab?

Or don't specify it at all, as this is the main tab.. co maybe just: Click the install..

> 2.  In SDK ENVIRONMENTS, click the Install button next to the [recomended](https://github.com/LipinskiPNordicSemi/chip-fork/blob/documentation/config/nrfconnect/.nrfconnect-recommended-revision) version of the nRF Connect SDK.

Reviewer: The official installation instructions in https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/installation/assistant.html#install-toolchain-manager uses "In SDK ENVIRONMENTS", so it's fine as is. But I would be fine with starting "Click the Install button..."

3.  A pop-up window will inform You about intallation directory. If You want to change the path click "Change directory" button.

> 4.  When the nRF Connect SDK is installed on your machine. The Install button changes to Open VS Code.

Reviewer: ```suggestion
4.  When the nRF Connect SDK is installed on your machine, the Install button changes to the Open VS Code button.
```

> 5.  Click the dropdown menu for the installed nRF Connect SDK version.

Reviewer: ```suggestion
5.  Click the dropdown menu next to the Open VS Code button for the installed nRF Connect SDK version.
```

> 6.  Select Open command prompt.

Reviewer 2: nit: maybe we can combine this point with previous one?

> 6.  Select Open command prompt.

Reviewer 2: I think it's called "Open terminal" on Linux if i'm not mistaken - can you please check?

Side note: Can we do sth like: `Open terminal` (adding ` character before and after) or **Open terminal** (adding two * to bold it).

> 7.  To make sure that ncs is up to date:

Reviewer: ```suggestion
7.  Make sure that all parts of the nRF Connect SDK are up to date:
```
How necessary is this step? Shouldn't the Toolchain Manager get everything set up?

Reviewer 2: It's more to ensure that the nRF Connect SDK revision is compatible with current Matter SDK (connectedhomeip).

So maybe.. "Make sure that nRF Connect SDK version is compatible with Matter SDK"

Reviewer: Could probably go even more specific:
```suggestion
7. Make sure that the nRF Connect SDK version is compatible with the Matter SDK version:
```

    ```

>          $ cd {connectedhomeip dirrectory}

Reviewer: ```suggestion
         $ cd {connectedhomeip directory}
```

    If you have the SDK already installed, continue to the next step and update
    the nRF Connect SDK after initializing environment variables.
4.  When the nRF Connect SDK is installed on your machine. The Install button changes to Open VS Code.

3.  Initialize environment variables referred to by the CHIP and the nRF Connect
    SDK build scripts. Replace _nrfconnect-dir_ with the path to your nRF
    Connect SDK installation directory, and _toolchain-dir_ with the path to GNU
    Arm Embedded Toolchain.
5.  Click the dropdown menu for the installed nRF Connect SDK version.

         $ source nrfconnect-dir/zephyr/zephyr-env.sh
         $ export ZEPHYR_TOOLCHAIN_VARIANT=gnuarmemb
         $ export GNUARMEMB_TOOLCHAIN_PATH=toolchain-dir
6.  Select Open command prompt.

4.  Update the nRF Connect SDK to the most recent supported revision by running
    the following command (replace _matter-dir_ with the path to Matter
    repository directory):
7.  To make sure that ncs is up to date:

         $ cd matter-dir
    ```
         $ cd {connectedhomeip dirrectory}
         $ python3 scripts/setup/nrfconnect/update_ncs.py --update

> Now you can proceed with the [Building](#building) instruction.

Reviewer 2: Should we leave this line :)
